### PR TITLE
Command to open CKBox image editing.

### PR DESCRIPTION
**Pull request by pszczesniak**
### Suggested merge commit message ([convention](https://ckeditor.com/docs/ckeditor5/latest/framework/contributing/git-commit-message-convention.html))

Internal(ckbox): New command responsible for opening a CKBox image editing dialog.
Closes cksource/ckeditor5-commercial#5634.
Closes https://github.com/cksource/ckeditor5-commercial/issues/5636.

---

### Additional information

**TOKEN\_URL = 'https://preview.ckbox.io/cke-workspaces-demo/?page=token'**

:point\_up: _This token endpoint should be used in manual test to test the CKBox Image Editor (getAsset wont throw any errors)._

### CODE REVIEW

**Comment by godai78** [comment]
I got the following error during local build:

```
./packages/ckeditor5-ckbox/src/ckboximageedit/ckboximageeditui.ts:10:0 - warning Invalid module name: "ckbox/ckboximageedit/imageiditui"

10    import { Plugin } from 'ckeditor5/src/core';
```

Shouldn't it be `imageeditui` rather?

**Comment by pszczesniak** [comment]
@godai78 :point\_right: [https://github.com/ckeditor/ckeditor5/pull/15209#issuecomment-1774450655](https://github.com/ckeditor/ckeditor5/pull/15209#issuecomment-1774450655) fixed.

**Review by arkflpc — CHANGES_REQUESTED** [review]

File: packages/ckeditor5-ckbox/src/augmentation.ts [hunk 1]
```diff
@@ -24,9 +27,20 @@ declare module '@ckeditor/ckeditor5-core' {
...
+		[ CKBoxImageEditEditing.pluginName ]: CKBoxImageEditEditing;
 	}
 
 	interface CommandsMap {
 		ckbox: CKBoxCommand;
+		ckboxImageEdit: CKBoxImageEditCommand;
 	}
 }
+
+declare global {
```
Comment (arkflpc) [inline packages/ckeditor5-ckbox/src/augmentation.ts:40]: cc: @filipsobol 
Do you know the consequences of moving it here for the projects in TypeScript that uses the editor? Will it interfere with something?

File: packages/ckeditor5-ckbox/src/ckboximageedit/ckboximageeditcommand.ts [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+	public override execute(): void {
+		this.fire<CKBoxImageEditorEvent<'open'>>( 'ckboxImageEditor:open' );
+	}
+
+	/**
+	 * Indicates if the CKBox Image Editor dialog is already opened.
+	 *
+	 * @private
+	 * @returns {Boolean}
+	 */
```
Comment (arkflpc) [inline packages/ckeditor5-ckbox/src/ckboximageedit/ckboximageeditcommand.ts:78]:
```suggestion
	 */
```

**Review by arkflpc — CHANGES_REQUESTED** [review]

**Review by arkflpc — APPROVED** [review]

File: packages/ckeditor5-ckbox/src/augmentation.ts (hunk 1 above)
Comment (filipsobol) [reply packages/ckeditor5-ckbox/src/augmentation.ts:40]: I don't expect any issues caused by moving this declaration to this file. I suspect that previous it didn't register at all (or not in all cases), so this is the right move :+1:

File: packages/ckeditor5-ckbox/src/ckboximageedit.ts [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+	 */
+	public static get pluginName() {
+		return 'CKBoxImageEdit' as const;
+	}
+
+	/**
+	 * @inheritDoc
+	 */
+	public static get requires() {
+		return [ CKBoxImageEditEditing, CKBoxImageEditUI ] as const;
```
Comment (niegowski) [inline packages/ckeditor5-ckbox/src/ckboximageedit.ts:30]: Does this feature require `CKBox` plugin?

File: packages/ckeditor5-ckbox/src/ckboximageedit/ckboximageeditui.ts [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+
+import { Plugin } from 'ckeditor5/src/core';
+import { ButtonView } from 'ckeditor5/src/ui';
+
+import ckboxImageEditIcon from '../../theme/icons/ckbox-image-edit.svg';
+
+/**
+ * The UI plugin of the CKBox image edit feature.
+ *
+ * It registers the `'imageEdit'` UI button in the editor's {@link module:ui/componentfactory~ComponentFactory component factory}
```
Comment (niegowski) [inline packages/ckeditor5-ckbox/src/ckboximageedit/ckboximageeditui.ts:18]:
```suggestion
 * It registers the `'ckboxImageEdit'` UI button in the editor's {@link module:ui/componentfactory~ComponentFactory component factory}
```

**Review by niegowski — CHANGES_REQUESTED** [review]
Partial review (without tests review).

File: packages/ckeditor5-ckbox/src/ckboximageedit.ts (hunk 3 above)
Comment (pszczesniak) [reply packages/ckeditor5-ckbox/src/ckboximageedit.ts]: Yes, it does.

File: packages/ckeditor5-ckbox/src/ckboximageedit.ts [hunk 5]
```diff
@@ -27,6 +28,6 @@ export default class CKBoxImageEdit extends Plugin {
 	 * @inheritDoc
 	 */
 	public static get requires() {
-		return [ CKBoxImageEditEditing, CKBoxImageEditUI ] as const;
+		return [ CKBox, CKBoxImageEditEditing, CKBoxImageEditUI ] as const;
```
Comment (niegowski) [inline packages/ckeditor5-ckbox/src/ckboximageedit.ts:31]: Maybe it should require `CKBoxEditing`? Wouldn't it be enough?

Comment (pszczesniak) [reply packages/ckeditor5-ckbox/src/ckboximageedit.ts]: Yes, i've checked and you're right.

File: packages/ckeditor5-ckbox/tests/ckboximageedit/ckboximageeditui.js [hunk 6]
```diff
@@ -0,0 +1,135 @@
...
+			// Header.
+			'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9',
+			// Payload.
+			btoa( JSON.stringify( { auth: { ckbox: { workspaces: [ 'workspace1' ] } } } ) ),
+			// Signature.
+			'signature'
+		].join( '.' );
+
+		window.CKBox = {
+			mountImageEditor: sinon.stub()
```
Comment (niegowski) [inline packages/ckeditor5-ckbox/tests/ckboximageedit/ckboximageeditui.js:35]: This looks like not been restored. I'd suggest using `testUtils.createSinonSandbox();` at the top level describe block.

File: packages/ckeditor5-ckbox/tests/ckboximageedit/ckboximageeditcommand.js [hunk 7]
```diff
@@ -0,0 +1,231 @@
...
+			btoa( JSON.stringify( { auth: { ckbox: { workspaces: [ 'workspace1' ] } } } ) ),
+			// Signature.
+			'signature'
+		].join( '.' );
+
+		domElement = global.document.createElement( 'div' );
+		global.document.body.appendChild( domElement );
+
+		window.CKBox = {
+			mountImageEditor: sinon.stub()
```
Comment (niegowski) [inline packages/ckeditor5-ckbox/tests/ckboximageedit/ckboximageeditcommand.js:38]: This looks like not been restored. I'd suggest using testUtils.createSinonSandbox(); at the top level describe block.

Comment (pszczesniak) [reply packages/ckeditor5-ckbox/tests/ckboximageedit/ckboximageeditcommand.js:40]: Done.

File: packages/ckeditor5-ckbox/tests/ckboximageedit/ckboximageeditui.js (hunk 6 above)
Comment (pszczesniak) [reply packages/ckeditor5-ckbox/tests/ckboximageedit/ckboximageeditui.js:38]: Done.

**Review by niegowski — APPROVED** [review]

**Pull request merged**